reactor-core/src/test/java/reactor/core/publisher/FluxUsingWhenTest.java

```diff
@@ -852,6 +855,180 @@ public void contextPropagationOnCancelWithNoHandler(Flux<String> source) {
 		assertThat(errorRef).hasValue(null);
 	}
 
+	// == tests checking callbacks don't pile up ==
+
+	@Test
+	public void noCancelCallbackAfterComplete() {
+		LongAdder cleanupCount = new LongAdder();
+		Flux<String> flux = Flux.usingWhen(Mono.defer(() -> Mono.just("foo")), Mono::just,
+				s -> Mono.fromRunnable(() -> cleanupCount.add(10)), //10 for completion
```

Comment:
WDYT about using strings, to make the test more readable?

Reply:
The numbers let us atomically detect application of several "callbacks". Using a `String` would make that property harder to get.

Reply:
`List<String>`?

`LongAdder` will successfully pass if we expect 100 (error) but actually called completion 10 times :D

Reply:
meh. the `List` is what I'm talking about: either skew operator /with a synchronized list or get potentially out of order / incomplete results.

the operator calling the callback 10 times when we only trigger completion 1 time would be quite a different problem...

Reply:
Okay... :D